From 8d38317f68d0e4663abbccdceca0bcd41f3dcf4a Mon Sep 17 00:00:00 2001
From: Patrick Hodoul <patrick.hodoul@autodesk.com>
Date: Mon, 27 Nov 2017 17:29:57 -0500
Subject: [PATCH 1/3] Fix compile error with Lut1DOp.cpp

Fix Linux compilation
Fix gcc 5.4.0 build breaks
Fix the temp filename for Linux
---
 src/core/Lut1DOp.cpp                 |  4 +++-
 src/core/MathUtils.cpp               | 23 ++++++++++++-----------
 src/pyglue/PyAllocationTransform.cpp |  1 -
 3 files changed, 15 insertions(+), 13 deletions(-)

diff --git a/src/core/Lut1DOp.cpp b/src/core/Lut1DOp.cpp
index 404ee0b..180e30f 100644
--- a/src/core/Lut1DOp.cpp
+++ b/src/core/Lut1DOp.cpp
@@ -188,7 +188,8 @@ OCIO_NAMESPACE_ENTER
         {
             return simple_lut[clamp(index, 0.0f, maxIndex)];
         }
-        
+
+#if defined(OCIO_UNIT_TEST) || !defined(USE_SSE)
         void Lut1D_Nearest(float* rgbaBuffer, long numPixels, const Lut1D & lut)
         {
             float maxIndex[3];
@@ -218,6 +219,7 @@ OCIO_NAMESPACE_ENTER
                 rgbaBuffer += 4;
             }
         }
+#endif
 #ifdef USE_SSE
         void Lut1D_Nearest_SSE(float* rgbaBuffer, long numPixels, const Lut1D & lut)
         {
diff --git a/src/core/MathUtils.cpp b/src/core/MathUtils.cpp
index e1a26c0..3881eb3 100644
--- a/src/core/MathUtils.cpp
+++ b/src/core/MathUtils.cpp
@@ -327,17 +327,6 @@ OCIO_NAMESPACE_ENTER
         GetV4Sum(vout, vout, v2);
     }
     
-    namespace
-    {
-    
-    void GetMxbResult(float* vout, float* m, float* x, float* v)
-    {
-        GetM44V4Product(vout, m, x);
-        GetV4Sum(vout, vout, v);
-    }
-    
-    } // anon namespace
-    
     bool GetMxbInverse(float* mout, float* vout,
                        const float* m_, const float* v_)
     {
@@ -372,6 +361,18 @@ OCIO_NAMESPACE_USING
 
 #include "UnitTest.h"
 
+namespace
+{
+
+    void GetMxbResult(float* vout, float* m, float* x, float* v)
+    {
+        GetM44V4Product(vout, m, x);
+        GetV4Sum(vout, vout, v);
+    }
+
+}
+
+
 OIIO_ADD_TEST(MathUtils, M44_is_diagonal)
 {
     {
diff --git a/src/pyglue/PyAllocationTransform.cpp b/src/pyglue/PyAllocationTransform.cpp
index 20bb50e..06b418a 100644
--- a/src/pyglue/PyAllocationTransform.cpp
+++ b/src/pyglue/PyAllocationTransform.cpp
@@ -53,7 +53,6 @@ OCIO_NAMESPACE_ENTER
         ///
         
         int PyOCIO_AllocationTransform_init(PyOCIO_Transform * self, PyObject * args, PyObject * kwds);
-        PyObject * PyOCIO_AllocationTransform_equals(PyObject * self,  PyObject * args);
         PyObject * PyOCIO_AllocationTransform_getAllocation(PyObject * self);
         PyObject * PyOCIO_AllocationTransform_setAllocation(PyObject * self,  PyObject * args);
         PyObject * PyOCIO_AllocationTransform_getNumVars(PyObject * self);
-- 
2.15.1

